bitnami/memcached/templates/service.yaml

```
@@ -8,7 +8,7 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.service.type }}
-  {{- if eq .Values.architecture "high-availability" }}
+  {{- if and (eq .Values.architecture "high-availability") (ne .Values.service.type "LoadBalancer") }}
```

We should also take into account the use case when `service.type=NodePort`. We could change the condition to something like:

```suggestion
  {{- if and (eq .Values.architecture "high-availability") (eq .Values.service.type "ClusterIP") }}
```

---

Yes, it will probably be more correct.